src/ristretto/bulletproofs_plus.rs

```diff
@@ -777,6 +777,72 @@ mod test {
         }
     }
 
+    #[test]
+    fn test_simple_aggregated_extended_proof() {
+        let mut rng = rand::thread_rng();
+        let bit_length = 64;
+
+        for extension_degree in [
+            CommitmentExtensionDegree::DefaultPedersen,
+            CommitmentExtensionDegree::AddOneBasePoint,
+        ] {
+            let factory = ExtendedPedersenCommitmentFactory::new_with_extension_degree(extension_degree).unwrap();
+
+            for aggregation_size in [2, 4] {
+                // 0.  Batch data
+                let mut proofs = vec![];
+                let mut statements_public = vec![];
+
+                #[allow(clippy::cast_possible_truncation)]
+                let (value_min, value_max) = (0u64, (1u128 << (bit_length - 1)) as u64);
```

Comment:
Nit: This isn't the full range of allowed values.

Reply:
```suggestion
                let (value_min, value_max) = (0u64, (1u128 << bit_length) - 1) as u64);
```

Reply:
This same construction is also used elsewhere in the same test file: [here](https://github.com/tari-project/tari-crypto/blob/abbd7208a4fef9664d233aff50a34a9637825dbe/src/ristretto/bulletproofs_plus.rs#L659) and [here](https://github.com/tari-project/tari-crypto/blob/abbd7208a4fef9664d233aff50a34a9637825dbe/src/ristretto/bulletproofs_plus.rs#L788) and [here](https://github.com/tari-project/tari-crypto/blob/abbd7208a4fef9664d233aff50a34a9637825dbe/src/ristretto/bulletproofs_plus.rs#L895).

However, the subsequent calls to `gen_range` are exclusive of the high value, so this will never cover the entire value range. This isn't problematic (since these are just randomized tests), but interesting to note.

The change you suggest (applied to the other instances as well) seems useful, so it's more clear to the reader what the intended value range is.

Reply:
Agreed, changed it all - good pickup, this was an oversight  :-)